### ISSUE TITLE
[keyvaluecache] Cannot use 'in' operator to search for 'Symbol(prefixesAreUnnecessaryForIsolation)' in bounded

### ISSUE THREAD

**Issue by junajan** [apollographql/apollo-utils#239]
Hi,
in the last [version](https://github.com/apollographql/apollo-utils/blob/%40apollo/utils.keyvaluecache%402.1.0/packages/keyValueCache/src/PrefixingKeyValueCache.ts#L50) of `@apollo/utils.keyvaluecache` package there is this line `return prefixesAreUnnecessaryForIsolationSymbol in c;` which breaks our apollo server instance (tested on node v16 and v18).

It throws this error:
```
TypeError: Cannot use 'in' operator to search for 'Symbol(prefixesAreUnnecessaryForIsolation)' in bounded
    at Function.prefixesAreUnnecessaryForIsolation (/app/node_modules/@apollo/server/node_modules/@apollo/utils.keyvaluecache/dist/PrefixingKeyValueCache.js:27:57)
    at new PrefixingKeyValueCache (/app/node_modules/@apollo/server/node_modules/@apollo/utils.keyvaluecache/dist/PrefixingKeyValueCache.js:9:36)
    at new ApolloServer (/app/node_modules/@apollo/server/dist/cjs/ApolloServer.js:110:28)
```

I was testing it on `@apollo/server` version `4.2.1` which [creates object](https://github.com/apollographql/apollo-server/blob/d7e9b97595b063f1e796ec4449850a16d19e8b18/packages/server/src/ApolloServer.ts#L303) `PrefixingKeyValueCache` like this:
```
new utils_keyvaluecache_1.PrefixingKeyValueCache(config.persistedQueries?.cache ?? this.cache, requestPipeline_js_1.APQ_CACHE_PREFIX),
```

With these params:
```
console.log(
  config.persistedQueries?.cache, // undefined
  this.cache, //  bounded
  requestPipeline_js_1.APQ_CACHE_PREFIX //  apq:
); 
```

Where constructor on [line 22](https://github.com/apollographql/apollo-utils/blob/%40apollo/utils.keyvaluecache%402.1.0/packages/keyValueCache/src/PrefixingKeyValueCache.ts#L22):
```
if (PrefixingKeyValueCache.prefixesAreUnnecessaryForIsolation(wrapped)) {
```

Uses mentioned function `prefixesAreUnnecessaryForIsolation` called with "bounded" string in params which turns into faulty code `Symbol(prefixesAreUnnecessaryForIsolation) in "bounded"`.

Is it a problem of `@apollo/server` or of `@apollo/utils.keyvaluecache`? With previous version of `utils.keyvaluecache` (v2.0.1) everything worked fine.

**Comment by trevor-scheer** [comment]
Hey @junajan, thanks for opening an issue and providing the details of the error.

`cache: "bounded"` is the default behavior for AS4 and you can just remove that configuration altogether.

However, we did run into this issue internally so @glasser merged a fix (https://github.com/apollographql/apollo-server/pull/7241) which re-adds support for `cache: "bounded"` in AS4.2.2.

So your 2 options are: drop the configuration (which I'd recommend) or update to latest AS version (or both!)

**Comment by glasser** [comment]
(the fix is actually v4.3.0)

**Comment by junajan** [comment]
Nice! Thank you both for such a quick response 👍 
And thank you for that idea with tsconfig, will give it a try.
